test/controllers/images_controller_test.rb

```diff
@@ -148,4 +161,27 @@ class ImagesControllerTest < ActionDispatch::IntegrationTest
       }
     }]
   end
+
+  test 'can get image submission form' do
+    get '/images/new'
+    assert_response :success
+    assert_select 'form/@action', images_path
+    assert_select 'form//input[@id="image_url"]', true
+    assert_select 'form//input[@id="image_tag_list"]', true
+  end
+
+  test 'can save valid image data' do
+    post images_url, params: { image: { url: 'https://example.com/foo.png', tag_list: 'baz, qux' } }
```

Comment:
FYI a common test you'll see in property:

```suggestion
    assert_difference -> { Image.count }, 1 do
      post images_url, params: { image: { url: 'https://example.com/foo.png', tag_list: 'baz, qux' } }
    end
```

It asserts that an image gets created in the database.

For the case when it fails there's also `assert_no_difference`

https://api.rubyonrails.org/classes/ActiveSupport/Testing/Assertions.html#method-i-assert_difference